### PR TITLE
About Medications page: In the event the API call fails, we need to p…

### PR DESCRIPTION
**Pull request by gerard-sullivan**
## Summary

- If the API fails, display the error message.
- Add the redux logic for the refill medication page.

## Related issue(s)

https://jira.devops.va.gov/browse/MHV-56950

## Testing done

- Tested locally

## Screenshots

_Note: This field is mandatory for UI changes (non-component work should NOT have screenshots)._

|         | Before | After |
| ------- | ------ | ----- |
| Mobile  |        |       |
| Desktop |        |   
Screenshot 2024-04-30 at 9 58 00 AM
    |

## What areas of the site does it impact?

Medications/Landing Page, List page, Detail page, Refill page

### Quality Assurance & Testing

- [ ] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

### :warning: Team Sites (only applies to modifications made to the VA.gov header) :warning:

- [ ] The vets-website header does not contain any web-components
- [ ] I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#local-dev) to test the injected header scenario
- [ ] I reached out in the `#sitewide-public-websites` Slack channel for questions

## Requested Feedback

(OPTIONAL) _What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

### CODE REVIEW

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 1]
```diff
@@ -513,40 +547,33 @@ const Prescriptions = () => {
...
+                  <div className="rx-page-total-info vads-u-border-color--gray-lighter" />
+                  <MedicationsList
+                    rxList={paginatedPrescriptionsList}
+                    pagination={pagination}
+                    selectedSortOption={selectedSortOption}
+                    updateLoadingStatus={updateLoadingStatus}
+                  />
+                </div>
+              ) : (
+                ''
```
Comment (agravell047) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:574]: using empty strings here isn't wrong, but its more semantically correct to use `<></>` or `null` when you don't want to render anything

File: src/applications/mhv/medications/actions/prescriptions.js [hunk 2]
```diff
@@ -39,15 +43,45 @@ export const clearAllergiesError = () => async dispatch => {
...
   dispatch({ type: Actions.Prescriptions.SET_DETAILS, prescription });
 };
 
+export const getRefillablePrescriptionsList = (
+  pageNumber,
+  sortEndpoint,
+) => async dispatch => {
+  try {
+    const response = await getRefillablePrescriptionList(
+      pageNumber,
```
Comment (agravell047) [inline src/applications/mhv/medications/actions/prescriptions.js:69]: there are no params for this function, so no need to pass in pageNumber or sortEndpoint

```
export const getRefillablePrescriptionList = () => {
  return apiRequest(
    `${apiBasePath}/prescriptions/list_refillable_prescriptions`,
    { headers },
  );
};
```

File: src/applications/mhv/medications/containers/Prescriptions.jsx [hunk 3]
```diff
@@ -168,6 +172,17 @@ const Prescriptions = () => {
...
+        if (prescriptionsApiError && isFirstRender) {
+          history.replace('/about');
+          return;
+        }
+        if (isFirstRender === undefined) {
+          setIsFirstRender(true);
+        } else {
+          setIsFirstRender(false);
+        }
+      }
```
Comment (agravell047) [inline src/applications/mhv/medications/containers/Prescriptions.jsx:185]: just curious, what the use case here?

**Review by agravell047 — CHANGES_REQUESTED** [review]

**Review by powellkerry — DISMISSED** [review]

File: src/applications/mhv/medications/containers/Prescriptions.jsx (hunk 3 above)
Comment (gerard-sullivan) [reply src/applications/mhv/medications/containers/Prescriptions.jsx:185]: @agravell047 I set the flag to redirect the page for the first API call only.

**Review by vmenshutin-bylight — DISMISSED** [review]

**Review by vmenshutin-bylight — DISMISSED** [review]

**Review by pjhill — DISMISSED** [review]

File: src/applications/mhv/medications/containers/RefillPrescriptions.jsx [hunk 4]
```diff
@@ -150,6 +149,25 @@ const RefillPrescriptions = ({ refillList = [], isLoadingList = true }) => {
...
+          categorizePrescriptions,
+          [[], []],
+        );
+        setFullRefillList(refillableList);
+        setFullRenewList(renewableList);
+      }
+      // disabled warning: fullRefillList must be left of out dependency array to avoid infinite loop
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    },
+    [refillablePrescriptionsList, prescriptionsApiError],
```
Comment (agravell047) [inline src/applications/mhv/medications/containers/RefillPrescriptions.jsx:168]: if this fails and we already have the list in state, it shows a half error state:
Screenshot 2024-05-07 at 5 24 00 PM

When I refresh the page:
Screenshot 2024-05-07 at 5 24 41 PM

**Review by agravell047 — CHANGES_REQUESTED** [review]

**Review by agravell047 — APPROVED** [review]

**Pull request merged**